Modularizing the code, using ui-router, demonstrating DRY tests, etc.

sanfordredlich opened the pull request:
Hi Kyle,

Ok, here are the changes we talked about. What do you think? 

There's an integration test added, btw, for a long-standing bug that happens if you add an item, switch to a view, then come back to "todo". There's also a fix for that now, which demonstrates Misko's comment that views referencing a model should always have a "dot", like "appData.todos". For some reason, though, it works by hand but the test fails. I'll look into that.

Cheers,
- Sanford

sanfordredlich committed:
switching to a feature based (and module-based) structure

sanfordredlich committed:
updated the docs to explain why modularization is useful

sanfordredlich committed:
gave credit to Josh Miller

sanfordredlich committed:
switched routing to use angular-ui-router; updated the documentation

sanfordredlich committed:
updated to use the latest brunch invocation format

sanfordredlich committed:
updated formatting

sanfordredlich committed:
removed unneeded partials from the top level since they were moved to the module folders

sanfordredlich committed:
fixed tests, made default route go to todo page

sanfordredlich committed:
integrated modularization and ui-router into 0.4.0 version of angular-brunch-seed; added test for problem where switching views resets the models (no complete fix yet, though it works by hand)

kylefinley commented:
Sanford,

Nice Job! It will take me a little while to go through everything, I'll try to get it merged over the weekend, though. 

Thanks again!

sanfordredlich commented:
Cool, thanks Kyle. Let me know if I can help with anything. I tested install, running server.sh and running both kinds of tests.  Cheers,  - S.

sanfordredlich commented:
Btw, on the last commit I just copied all of angular-brunch-seed into my fork, then updated the necessary files individually, to avoid conflicts with your latest code.  - S.

kylefinley commented:
Hey @sanfordredlich,

Every thing looks good so far -- the only issue I need to resolve is with the partials compiling to separate files, but  I think that's an issue that can be solved in `jade-angularjs-brunch`.

Considering that the core routing infrastructure will be removed from Angular 1.2. I'll probably wait to merge with master until after Angular 1.2 is released, which should be soon. I just wanted to give you and update, so you didn't think I forgot about it.

Commit:
reverted breaking changes so the app works again

mattallty committed:
Add --server argument to launch the server

The documentation states that one just need to run `./scripts/development.sh` and open browser to `http://localhost:3333`, but we need to pass --server argument to the `brunch watch` command to effectively launch the server.

sanfordredlich committed:
Merge pull request #2 from mattallty/master

Add --server argument to launch the server
